__tests__/unit-tests/existingRegionMiddleware.test.js

```diff
@@ -0,0 +1,35 @@
+const existingRegionMiddleware = require('../../src/middlewares/existingRegionMiddleware');
+const path = require('path');
+
+describe('existingRegionMiddleware', () => {
+    const res = {};
+    const next = jest.fn();
+
+    res.status = jest.fn().mockReturnValue(res);
+    res.sendFile =  jest.fn()
+
+    it.each([
+        [{params: {regionId: "asia"}}],
+        [{params: {regionId: "americas"}}],
+        [{params: {regionId: "europe"}}],
+    ])('should pass control to the next middleware when a valid region Id has been sent in the request', (req) => {
+        existingRegionMiddleware(req, res, next);
+        expect(next).toHaveBeenCalled()
+    });
+
+    it.each([
+        [{params: {regionId: "africa"}}],
+        [{params: {regionId: "australia & oceania"}}],
+        [{params: {regionId: ""}}],
```

Comment:
I will need to make a mental note to advise anyone adding sources from these regions to alter this test. Very good call though.